cmd/kube-proxy/app/server_linux_test.go

```diff
@@ -719,3 +721,221 @@ func TestProxyServer_platformSetup(t *testing.T) {
 		})
 	}
 }
+
+type fakeConntracker struct {
+	called []string
+	err    error
+}
+
+func (fc *fakeConntracker) SetMax(ctx context.Context, max int) error {
+	fc.called = append(fc.called, "SetMax")
+	return fc.err
+}
+func (fc *fakeConntracker) SetTCPEstablishedTimeout(ctx context.Context, seconds int) error {
+	fc.called = append(fc.called, "SetTCPEstablishedTimeout")
+	return fc.err
+}
+func (fc *fakeConntracker) SetTCPCloseWaitTimeout(ctx context.Context, seconds int) error {
+	fc.called = append(fc.called, "SetTCPCloseWaitTimeout")
+	return fc.err
+}
+func (fc *fakeConntracker) SetTCPBeLiberal(ctx context.Context, value int) error {
+	fc.called = append(fc.called, "SetTCPBeLiberal")
+	return fc.err
+}
+func (fc *fakeConntracker) SetUDPTimeout(ctx context.Context, seconds int) error {
+	fc.called = append(fc.called, "SetUDPTimeout")
+	return fc.err
+}
+func (fc *fakeConntracker) SetUDPStreamTimeout(ctx context.Context, seconds int) error {
+	fc.called = append(fc.called, "SetUDPStreamTimeout")
+	return fc.err
+}
+
+func TestSetupConntrack(t *testing.T) {
+	_, ctx := ktesting.NewTestContext(t)
+	tests := []struct {
+		name         string
+		config       proxyconfigapi.KubeProxyConntrackConfiguration
+		expect       []string
+		conntrackErr error
+		wantErr      bool
+	}{
+		{
+			name: "do nothing if conntrack config is empty",
+			config: proxyconfigapi.KubeProxyConntrackConfiguration{
+				MaxPerCore:            nil,
+				Min:                   nil,
+				TCPEstablishedTimeout: nil,
+				TCPCloseWaitTimeout:   nil,
+				TCPBeLiberal:          false,
+				UDPTimeout:            metav1.Duration{Duration: 0},
+				UDPStreamTimeout:      metav1.Duration{Duration: 0},
+			},
```

Comment:
since the test name literally calls out "config is empty", you could just use

    config: proxyconfigapi.KubeProxyConntrackConfiguration{},

likewise, you could just not specify all the defaulted fields in the other test cases?

Also (and maybe contradictorily), I feel like an explicit `expect: nil` or `expect: []string{}` would be clearer on the tests where you expect no conntrack calls?

Reply:
done